src/main/java/org/apache/commons/imaging/formats/tiff/TiffImageParser.java

```diff
@@ -75,7 +75,7 @@ private Rectangle checkForSubImage(
         // dimensions of the image that is being read.  This method
         // returns the sub-image specification, if any, and leaves
         // further tests to the calling module.
-        if (params.isSubImageSet()) {
+        if (params != null && params.isSubImageSet()) {
```

Comment:
Hadn't time to review properly, but it would be useful to check if the method that calls this private method couldn't be updated instead, to avoid a null object reaching here. Thanks

Reply:
Thanks for the quick response!
There are multiple calling methods and IMHO it makes the most sense within this method, so that any consumers can just use checkForSubImage(params) to find out whether there's any subImage data to be extracted no matter the given params.

Reply:
The only thing missing here is a unit test that shows why this is a problem.

Reply:
This is an actual problem. If you want to read a multipage tiff with:

`TiffImageParser().getAllBufferedImages(File(path))`

you will run into a npe.

Reply:
@Ditscheridou, @mercuryLuz 
This PR still needs a test.